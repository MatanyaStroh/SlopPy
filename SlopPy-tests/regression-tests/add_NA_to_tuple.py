# try to add an NA object to a tuple in various ways;
# all attempts should get rejected

x = 1 / 0
assert type(x) is NA

# tuple constructor should ignore all instances of x
assert (x,) == ()
assert (x,x) == ()
assert (x,x,x) == ()
assert (1,2,x,3,4) == (1,2,3,4)
assert (1,2,x,3,4,x) == (1,2,3,4)
assert (x,1,2,x,3,4,x) == (1,2,3,4)
assert (x,x,x,x,x,1,x) == (1,)

