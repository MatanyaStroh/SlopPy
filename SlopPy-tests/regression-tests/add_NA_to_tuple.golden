{NA <type 'exceptions.ZeroDivisionError'> 'integer division or modulo by zero'
 Traceback (most recent call last):
  File "regression-tests/add_NA_to_tuple.py", line 4, in <module>
    x = 1 / 0

Locals: {'__builtins__': <module '__builtin__' (built-in)>, '__name__': '__main__', '__file__': 'regression-tests/add_NA_to_tuple.py', '__doc__': None, '__package__': None}
}
{BUILD_TUPLE(* NA *) | regression-tests/add_NA_to_tuple.py <module>:8}
{BUILD_TUPLE(* NA *) | regression-tests/add_NA_to_tuple.py <module>:9}
{BUILD_TUPLE(* NA *) | regression-tests/add_NA_to_tuple.py <module>:10}
{BUILD_TUPLE(* NA *) | regression-tests/add_NA_to_tuple.py <module>:11}
{BUILD_TUPLE(* NA *) | regression-tests/add_NA_to_tuple.py <module>:12}
{BUILD_TUPLE(* NA *) | regression-tests/add_NA_to_tuple.py <module>:13}
{BUILD_TUPLE(* NA *) | regression-tests/add_NA_to_tuple.py <module>:14}
